File: apps/api/src/JobTracker.Domain/Entities/Job.cs
using JobTracker.Domain.Common;
using JobTracker.Domain.Enums;

namespace JobTracker.Domain.Entities;

public class Job : BaseEntity
{
    public Guid UserId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Company { get; set; } = string.Empty;
    public string? Url { get; set; }
    public string? Description { get; set; }
    public string? Salary { get; set; }
    public string? Location { get; set; }
    public string? Note { get; set; }
    public string? Source { get; set; }
    public JobStatus Status { get; set; } = JobStatus.Applied;

    // Navigation properties
    public User? User { get; set; }
    public ICollection<Interview> Interviews { get; set; } = new List<Interview>();
}

File: apps/web/src/types/job.ts
export type StageId =
  | 'applied'
  | 'test'
  | 'interview'
  | 'offer'
  | 'rejected'
  | 'ghosted';

export type InterviewStatus = 'passed' | 'failed' | 'upcoming';

export interface Interview {
  id: string;
  round: string;
  date: string;
  status: InterviewStatus;
  notes?: string;
}

export interface Job {
  id: string;
  company: string;
  title: string;
  location: string;
  salary?: string;
  appliedAt?: string;
  stage: StageId;
  source?: string;
  jdLink?: string;
  note?: string;
  interviews: Interview[];
}

export const locationLabel: Record<string, string> = {
  Onsite: 'Trực tiếp',
  Hybrid: 'Linh hoạt',
  Remote: 'Từ xa',
};